lightning/src/ln/msgs.rs

```diff
@@ -4338,6 +4359,64 @@ mod tests {
 		} else { panic!(); }
 	}
 
+	#[test]
+	fn encoding_final_onion_hop_data_with_trampoline_packet() {
```

Comment:
Are these tests from the spec? Would be good to have some high level comments for them